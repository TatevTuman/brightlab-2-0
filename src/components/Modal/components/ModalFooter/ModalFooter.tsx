import React, { memo } from 'react'
import { Children } from '@types'
import styles from '../../Modal.module.scss'

export interface ModalFooterProps {
  children: Children
}

type ModalFooterType = React.FC<ModalFooterProps>

const ModalFooter: ModalFooterType = props => {
  return <footer className={styles.modalFooter}>{props.children}</footer>
}

export default memo(ModalFooter)
