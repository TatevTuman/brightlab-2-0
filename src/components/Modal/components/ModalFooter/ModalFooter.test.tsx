import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import ModalFooter from './ModalFooter'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('ModalFooter', () => {
  it('renders correctly', async () => {
    const { container, getByText } = render(<ModalFooter>children</ModalFooter>)
    const awaitedContainer = await waitFor(() => container)

    expect(awaitedContainer).toMatchSnapshot()
    expect(getByText('children')).toBeInTheDocument()
  })
})
