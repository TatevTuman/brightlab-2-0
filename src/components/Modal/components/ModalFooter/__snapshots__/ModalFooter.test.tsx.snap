// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter renders correctly 1`] = `
<div>
  <footer
    class="modalFooter"
  >
    children
  </footer>
</div>
`;
